hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/BlockManagerImpl.java

```diff
@@ -162,7 +157,13 @@ public long persistPutBlock(KeyValueContainer container,
           }
         }
 
-        db.getStore().putBlockByID(batch, incrementalEnabled, localID, data,
+        if (incrementalEnabled.get() && !VersionedDatanodeFeatures.isFinalized(
+            HDDSLayoutFeature.HBASE_SUPPORT)) {
+          LOG.warn("DataNode has not finalized upgrading to a version that " +
+              "supports incremental chunk list. Fallback to full chunk list");
+          incrementalEnabled.set(false);
```

Comment:
There is chance that if pubBlock is first called before HDDSLayoutFeature.HBASE_SUPPORT is finalized, then incrementalEnabled will be set as false. In this case, incrementalEnabled  will not come back as "true" without the DN restart.

Reply:
@jojochuang , while going through the code, I found we have two properties currently for incremental chunk list, 
"ozone.client.incremental.chunk.list" // client side configure, default is false. For Hbase ozone client, this should be true. For other application ozone client, this can be false. 
"ozone.incremental.chunk.list"   // server side configure.   Can we remove this property? Because it's default value is false, once HBASE_SUPPORT finalized, user needs to change it's value to true, and then restart the DN for that. Once "HDDSLayoutFeature.HBASE_SUPPORT" feature is finalized, we expect this property always be true, right?  So If it can be removed, then we only need to check whether to reject a partial chunk list based on  "HDDSLayoutFeature.HBASE_SUPPORT" feature is finalized or not on DN side. 

BTW,  there is one case, client sends a partial chunk list, server side  "HDDSLayoutFeature.HBASE_SUPPORT" feature is not finalized yet, then this client request should fail, looks like DatanodeStoreWithIncrementalChunkList#putBlockByID doesn't cover this case yet.